eiffel-syntax-and-usage/the-links-object.md

```diff
@@ -75,7 +75,7 @@ __Description:__ Identifies the composition from which an artifact declared by [
 
 ### ENVIRONMENT
 __Required in:__ None  
-__Optional in:__ [EiffelArtifactCreatedEvent](../eiffel-vocabulary/EiffelArtifactCreatedEvent.md), 
+__Optional in:__ [EiffelArtifactCreatedEvent](../eiffel-vocabulary/EiffelArtifactCreatedEvent.md), [EiffelTestSuiteStartedEvent](../eiffel-vocabulary/EiffelTestSuiteStartedEvent.md), 
```

Comment:
Yes I think we want it that way. It relates to the discussion of what a test suite really is, which we haven't finalized yet as far as I'm aware.
The scenario I have in mind is that the execution of a TERC is divided into several steps:
1. For each recipe in the TERC, which might have different test environment needs, we will provision and deploy a test environment. This will render an EiffelEnvironmentDefinedEvent.
2. The test cases in each recipe, forming a test suite, is then sent to a test executor together with a reference to the created environment to test on.
3. The executor initiates its test execution by sending an EiffelTestSuiteStarted event referencing the given environment, and then executes each test case in the given test suite.
Of course it would be awkward if the test case referenced an orthogonal environment to what its test suite did, but that could be handled by rules and guidelines.
I see the need to discuss this matter in front of a whiteboard, and I look forward to get that possibility in one of the coming weeks.